### ISSUE TITLE
[D3D9] Screen shaking during FMV in all Hatsune Miku games

### ISSUE THREAD

**Issue by Leopard20** [hrydgard/ppsspp#13374]
The screen shakes when playing the intro videos.
[pd.zip](https://github.com/hrydgard/ppsspp/files/5167870/pd.zip)
I'm not sure when this started happening, but I'm running the latest dev build (v1.10.3-550)

I'm not sure if this is related or not:
https://github.com/hrydgard/ppsspp/issues/13370

**Comment by hrydgard** [comment]
#13370 is likely related yes. The new framebuffer management that helped fix Test Drive needs a bit more work.

**Comment by hrydgard** [comment]
Is this ok now after #13397 ?

**Comment by leopard20** [comment]
Nope. Not fixed yet.
Not sure if this is useful:

**Comment by hrydgard** [comment]
This only happens in D3D9, right?

**Comment by leopard20** [comment]
I'm on Android! ¯\_(ツ)_/¯
Vulkan and OpenGL

**Comment by hrydgard** [comment]
Oh! Weird.

**Comment by leopard20** [comment]
Good guess! Yeah. I removed the PP effects and it now looks fine.
I think both of "natural colors" effects have this issue. The rest looked fine.

**Comment by hrydgard** [comment]
I'm guessing the game somehow ends up drawing the video in two different ways, one frame directly to the backbuffer and the other through a framebuffer, and it flips between those and there's a subtle offset bug in one of them when using postprocessing ...

**Comment by leopard20** [comment]
You were right about D3D9 though. It is broken with and without the shader. So please add the tag again.
I fixed the issue with one of the shaders.

**Comment by unknownbrackets** [comment]
> I'm guessing the game somehow ends up drawing the video in two different ways, one frame directly to the backbuffer and the other through a framebuffer, and it flips between those and there's a subtle offset bug in one of them when using postprocessing ...

Yes, as per #3417.  Maybe this is related to DrawFramebufferToOutput when it goes directly through presentation?

-[Unknown]

**Comment by hrydgard** [comment]
Yes, that's what I'd guess.

**Comment by Unknowncat1991** [comment]
D3D9 is now gone https://github.com/hrydgard/ppsspp/pull/19951
